service/orchestrator/metrics.go

```diff
@@ -188,15 +196,12 @@
 		return nil, status.Errorf(codes.InvalidArgument, "%v", newError)
 	}
 
-	// Lock metrics mutex
-	svc.mm.Lock()
-	defer svc.mm.Unlock()
-
 	// Check, if metric exists according to req.MetricId
-	if metric, ok = svc.metrics[req.MetricId]; !ok {
-		newError := fmt.Errorf("metric with identifier %s does not exist", req.MetricId)
-		log.Error(newError)
-		return nil, status.Errorf(codes.NotFound, "%v", newError)
+	err = svc.storage.Get(&metric, "id = ?", req.MetricId)
```

Comment:
use `svc.count`?

Reply:
I think there should not be any significant gains in getting vs. counting one entry. Get seemed a bit more natural to me. You will probably point me to a line of code where I did use count before ;) However, this will be anyway gone once we do proper GORM associations I think.

Reply:
Yes, you  told me once to use count for efficiency reasons :D